scipy/interpolate/_rgi.py

```diff
@@ -332,7 +332,7 @@ def __call__(self, xi, method=None):
         elif method in self._SPLINE_METHODS:
             if is_method_changed:
                 self._validate_grid_dimensions(self.grid, method)
-            result = self._evaluate_spline(self.values.T, xi, method)
+            result = self._evaluate_spline(self.values, xi, method)
```

Comment:
This transpose is too eager. Consider a 4D problem with `values.shape == (2, 3, 4, 5, 6)`. The `.T` converts it into (6, 5, 4, 3, 2) which for a 4D problem is not right, it should be (5, 4, 3, 2, 6)